server/monitor.go

```diff
@@ -834,30 +834,95 @@ func (s *Server) HandleStacksz(w http.ResponseWriter, r *http.Request) {
+	ClientConnectURLs []string          `json:"connect_urls,omitempty"`
+	MaxConn           int               `json:"max_connections"`
+	MaxSubs           int               `json:"max_subscriptions,omitempty"`
+	PingInterval      time.Duration     `json:"ping_interval"`
+	MaxPingsOut       int               `json:"ping_max"`
+	HTTPHost          string            `json:"http_host"`
+	HTTPPort          int               `json:"http_port"`
+	HTTPSPort         int               `json:"https_port"`
+	AuthTimeout       float64           `json:"auth_timeout"`
+	MaxControlLine    int32             `json:"max_control_line"`
+	MaxPayload        int               `json:"max_payload"`
+	MaxPending        int64             `json:"max_pending"`
+	Cluster           ClusterOptsVarz   `json:"cluster,omitempty"`
+	Gateway           GatewayOptsVarz   `json:"gateway,omitempty"`
+	LeafNode          LeafNodeOptsVarz  `json:"leaf,omitempty"`
+	TLSTimeout        float64           `json:"tls_timeout"`
+	WriteDeadline     time.Duration     `json:"write_deadline"`
+	Start             time.Time         `json:"start"`
+	Now               time.Time         `json:"now"`
+	Uptime            string            `json:"uptime"`
+	Mem               int64             `json:"mem"`
+	Cores             int               `json:"cores"`
+	CPU               float64           `json:"cpu"`
+	Connections       int               `json:"connections"`
+	TotalConnections  uint64            `json:"total_connections"`
+	Routes            int               `json:"routes"`
+	Remotes           int               `json:"remotes"`
+	InMsgs            int64             `json:"in_msgs"`
+	OutMsgs           int64             `json:"out_msgs"`
+	InBytes           int64             `json:"in_bytes"`
+	OutBytes          int64             `json:"out_bytes"`
+	SlowConsumers     int64             `json:"slow_consumers"`
+	Subscriptions     uint32            `json:"subscriptions"`
+	HTTPReqStats      map[string]uint64 `json:"http_req_stats"`
+	ConfigLoadTime    time.Time         `json:"config_load_time"`
+}
+
+// ClusterOptsVarz contains monitoring cluster information
+type ClusterOptsVarz struct {
+	Host        string  `json:"addr,omitempty"`
+	Port        int     `json:"cluster_port,omitempty"`
+	AuthTimeout float64 `json:"auth_timeout,omitempty"`
+}
+
+// GatewayOptsVarz contains monitoring gateway information
+type GatewayOptsVarz struct {
+	Name           string                  `json:"name,omitempty"`
+	Host           string                  `json:"host,omitempty"`
+	Port           int                     `json:"port,omitempty"`
+	AuthTimeout    float64                 `json:"auth_timeout,omitempty"`
+	TLSTimeout     float64                 `json:"tls_timeout,omitempty"`
+	Advertise      string                  `json:"advertise,omitempty"`
+	ConnectRetries int                     `json:"connect_retries,omitempty"`
+	Gateways       []RemoteGatewayOptsVarz `json:"gateways,omitempty"`
```

Comment:
Not sure if we should rather have array of pointers here or not. The way it is right now users don't need to check if Gateways is nil or not, and this field does not show if the array is empty:
```
  "gateway": {
    "name": "A",
    "host": "0.0.0.0",
    "port": 64069,
    "auth_timeout": 1,
    "tls_timeout": 0.5
  }
```
Here is when there is one remote:
```
"gateway": {
    "name": "A",
    "host": "0.0.0.0",
    "port": 64078,
    "auth_timeout": 1,
    "tls_timeout": 0.5,
    "gateways": [
      {
        "name": "B",
        "urls": [
          "localhost:1234"
        ]
      }
    ]
  }
```

Reply:
I think its fine. What would be the advantage of a pointer in this context?

Reply:
None that I can think of, was just wondering if your preference was to have array of pointers here.